### ISSUE TITLE
SequenceCollection's degap method will fail unless the underlying sequence objects are DNA

### ISSUE THREAD

**Issue by ElDeveloper** [biocore/scikit-bio#1036]
The `degap` method will fail if you read in sequences and don't cast them explicitly to `DNA` objects:

```python-traceback

In [27]: from skbio import SequenceCollection

In [28]: plel = SequenceCollection.read('97_otus.fasta')

In [29]: plel.degap()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-29-adb4ed174cf0> in <module>()
----> 1 plel.degap()

/Users/yoshikivazquezbaeza/git_sw/scikit-bio/skbio/alignment/_alignment.pyc in degap(self)
    382 
    383         """
--> 384         return SequenceCollection([seq.degap() for seq in self])
    385 
    386     @experimental(as_of="0.4.0")

AttributeError: 'Sequence' object has no attribute 'degap'
```

**Comment by jairideout** [comment]
Thanks for reporting this @ElDeveloper! We are aware of the issue, this was introduced when stabilizing the `skbio.sequence` API. `SequenceCollection` and `Alignment` are next up for redesign so we'll be sure this gets resolved then.

**Comment by ebolyen** [comment]
fixed by removing SequenceCollection #1220
